CRM/Contribute/BAO/Contribution.php

```diff
@@ -2959,6 +2959,10 @@ public static function recordFinancialAccounts(&$params, $financialTrxnValues =
         $balanceTrxnParams['status_id'] = $statusId;
         $balanceTrxnParams['payment_instrument_id'] = $params['contribution']->payment_instrument_id;
         $balanceTrxnParams['check_number'] = CRM_Utils_Array::value('check_number', $params);
+        if (!empty($balanceTrxnParams['from_financial_account_id']) &&
```

Comment:
So, I would do a separate PR with just the changes to set 'is_payment' (generally with a BAO level test at the same time)